src/ffi3/datetime.rs

```diff
@@ -0,0 +1,409 @@
+use ffi3::object::*;
+use ffi3::pycapsule::PyCapsule_Import;
+use ffi3::pyport::Py_hash_t;
+use std::ffi::CString;
+use std::option::Option;
+use std::os::raw::{c_char, c_int, c_uchar};
+
+#[cfg_attr(windows, link(name = "pythonXY"))]
+extern "C" {
+    pub static mut PyDateTime_DateType: PyTypeObject;
+    pub static mut PyDateTime_TimeType: PyTypeObject;
+    pub static mut PyDateTime_DateTimeType: PyTypeObject;
+
+    pub static mut PyDateTime_DeltaType: PyTypeObject;
+    pub static mut PyDateTime_TZInfoType: PyTypeObject;
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct PyDateTime_CAPI {
+    pub DateType: *mut PyTypeObject,
+    pub DateTimeType: *mut PyTypeObject,
+    pub TimeType: *mut PyTypeObject,
+    pub DeltaType: *mut PyTypeObject,
+    pub TZInfoType: *mut PyTypeObject,
+    #[cfg(Py_3_7)]
+    pub TimeZone_UTC: *mut PyObject,
+
+    pub Date_FromDate: Option<
```

Comment:
It's been a while since I wrote this code, but I think the reason is that this structure is a global starts out empty and is only initialized later by a call to `PyDateTime_IMPORT()`, so during the pre-import phase all of these pointers would be to uninitialized memory.

I think I actually solved this a bit more elegantly by using `lazy_static` for the global API module, so that the global is initialized on first use.

Having each of them be nullable *seems* like the right thing to do (since they are nullable and null until imported), though I don't know if I've actually accomplished that properly in this case, or if it matters when the supported way to use this structure is to access the static global, and in the static global it's not likely that these would not be populated.

Reply:
It's a struct that is never constructed in rust code (just casted to from the import function), so I'd say there's no dangling pointer there. The `lazy_static` should be sufficient.